### PR TITLE
feat(server): capture other constructor arguments and forward to base class

### PR DESCRIPTION
**Pull request by mmccartn**
Allows uses to instantiate the base WebSocketServer from an existing `http` server. This is motivated by the desire to host a `ws-plus`  server and an `express` server from the same network port.

Example:
```js
const { Server: WSServer } = require('ws-plus')
const express = require('express')
const http = require('http')

// create an express app
const expressApp = express()
expressApp.get('/', (req, res) => res.send('Hello World'))

// create a root http server context
const server = http.createServer()
server.on('request', expressApp)

// create a ws-plus server
const wsClients = new WSServer({ server })

// host http and wss servers from the same network port
server.listen(54321)
```

Or, with `fasatify`:
```js
const fastify = require('fastify')()
fastify.get('/', (req, reply) => reply.send('Hello World'))
const wsClients = new WSServer({ server: fastify.server })
fastify.listen(54321)
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -289,6 +289,7 @@ Options:
  - `maxSendBuffer`: Size of the send buffer in bytes. This governs how much the server can queue to an unacknowledging recipient (for example, during slow connections) before giving up. Messages that exceed the send buffer are dropped. (default: `20000`)
  - `rootObject`: Serialize messages as JSON objects instead of JSON arrays. Useful when handling clients that don't support top-level arrays. (default: `false`)
  - `verbosity`: Show more/less log messages (default `1`). Use `0` to silence.
+ - `wssOpts`: Pass all other arguments to the base `WebSocketServer` constructor.
```
Comment (foxxyz) [inline README.md:292]:
```suggestion
 - Any other option allowed in [`ws.WebSocketServer`](https://github.com/websockets/ws/blob/master/doc/ws.md#new-websocketserveroptions-callback)
```

**Comment by foxxyz** [comment]
Running your example, it's a bit weird to see the following printed out (even if `server.listen(...)` is never called).

Perhaps we need to move the notification to be triggered by the [`listening` event](https://github.com/websockets/ws/blob/master/doc/ws.md#event-listening) instead of in the constructor, and use [`server.address()`](https://github.com/websockets/ws/blob/master/doc/ws.md#serveraddress) to get the correct host/port.

```
> ./index.js
Serving websocket server at ws://127.0.0.1:null. Awaiting clients...
```

File: test/server.test.js [hunk 2]
```diff
@@ -38,6 +39,19 @@ describe('Server Creation', () => {
...
+        expect(server.server._server).toBe(httpServer)
+        expect(server.server.options.port).toBe(null)
+    })
+    it('prints square brackets around ipv6 addresses when listening', async() => {
+        const info = jest.spyOn(console, 'info')
+        const httpServer = http.createServer()
+        server = new Server({ server: httpServer })
+        await new Promise(res => httpServer.listen({ port: 54322 }, res))
+        expect(info).toHaveBeenCalledWith('Serving websocket server at ws://[::]:54322. Awaiting clients...')
+    })
```
Comment (foxxyz) [inline test/server.test.js:54]: This test is currently hanging CI because I think it's not actually closing the http server on test completion.

**Review by foxxyz — CHANGES_REQUESTED** [review]

File: test/server.test.js (hunk 2 above)
Comment (mmccartn) [reply test/server.test.js:54]: I assumed WebSocketServer would close it, [but that is not the case when the server context is an options input](https://github.com/websockets/ws/blob/master/lib/websocket-server.js#L168)

**Pull request merged**